resources/troubleshooting-guides/push_notification_troubleshooting.md

```diff
@@ -0,0 +1,117 @@
+   
+> For more information about configuration, see [Channel specific application configuration in Adobe Campaign.](https://helpx.adobe.com/campaign/kb/configuring-app-sdk.html#ChannelspecificapplicationconfigurationinAdobeCampaign).  
+  
+## To set up Android/iOS app for receiving push notifications  
+
+### To set up Android app  
+
+   1. Create the Firebase Messaging Service and add it to the Android Manifest file.  
+   For more information, see [Set up a Firebase Cloud Messaging client app on Android](https://firebase.google.com/docs/cloud-messaging/android/client)
+   2. Verify this step in Project Griffon.
+   For more information, see [Set up Project Griffon](https://aep-sdks.gitbook.io/docs/beta/project-griffon/set-up-project-griffon) and [Use Project Griffon](https://aep-sdks.gitbook.io/docs/beta/project-griffon/using-project-griffon).  
+   3. Generate a push token for app using FireBase Instance Id class.  
+   4. Set the push token to the Expereince Platform SDK by calling Set [Push Identifier](https://aep-sdks.gitbook.io/docs/using-mobile-extensions/adobe-campaign-standard/adobe-campaign-standard-api-reference#set-up-push-messaging).     
+
+### To set up iOS app  
+  
+  1. After you launch the app, register the app to receive the APNS token by calling `registerForRemoteNotifications`.  
+  This API generates the APNS token and returns token through the delegate function `didRegisterForRemoteNotificationsWithDeviceToken`.    
+  For more information, see [Registering Your App with APNs](https://developer.apple.com/documentation/usernotifications/registering_your_app_with_apns?language=objc)
+  2. In `didRegisterForRemoteNotificationsWithDeviceToken`, set the Push identifier by calling the `setPushIdentifier`.  
+  For more information, see [setPushIdentifier](https://aep-sdks.gitbook.io/docs/using-mobile-extensions/adobe-analytics-mobile-services#set-up-push-messaging).
+ 
+### To verify that push token is successfully synced with ECID service, complete the following steps:
+
+{% hint style="warning" %}
+If SDK privacy status is`optedout`, the push identifier will not be set.
+{% endhint %}
+
+   1. Verify push token sync with the Experience Cloud Identity service.  
+   2. To verify Launch your app with the device connected to a Project Griffon session.   
+   3. In the list of events, verify that you have an event with type _SetPushIdentifier_. In the details panel on the right, verify the value of the push token for this device. The value in pushIdentifier is the same value that is sent to the ECID service. For more information, see screenshot below.
+![alt text](../../.gitbook/assets/push_token_to_identity.png "Verify SetPushIdentifier event.")
+   2. Verify that app’s push token is in the Campaign instance that was mapped with the correct Experience cloud id(mid) by clicking _Adobe Campaign -> Administrator -> Channels -> Mobile App (AEP SDK)_.
```

Comment:
- Please put the menu items in bold.

- Please reword as follows:
"To verify that app’s push token that was mapped to the correct Experience cloud id(mid) is in the Campaign instance, click  **Adobe Campaign** -> **Administrator** -> **Channels** -> **Mobile App (AEP SDK)**.